core/src/sumeragi/main_loop.rs

```diff
@@ -1039,21 +1041,31 @@ pub(crate) fn run(
                 },
             );
 
-        process_message_independent(
-            sumeragi,
-            &mut state,
-            &mut voting_block,
-            current_view_change_index,
-            &mut view_change_proof_chain,
-            &round_start_time,
-            is_genesis_peer,
-        );
+        if is_genesis_peer {
+            process_message_independent::<true>(
+                sumeragi,
+                &mut state,
+                &mut voting_block,
+                current_view_change_index,
+                &mut view_change_proof_chain,
+                &round_start_time,
+            );
+        } else {
+            process_message_independent::<false>(
+                sumeragi,
+                &mut state,
+                &mut voting_block,
+                current_view_change_index,
+                &mut view_change_proof_chain,
+                &round_start_time,
+            );
+        }
```

Comment:
I think `is_genesis_peer` variable confused you here. It's used only for tests in soft fork detection. We shouldn't apply genesis round rules for such peer.

Reply:
It should be named something like `peer_with_soft_fork`.

Reply:
is it better now?

Reply:
Yes, feel free to resolve.